### PR TITLE
Refactor Index item hierarchy

### PR DESCRIPTION
**Pull request by temanbrcom**

### CODE REVIEW

File: server/src/main/java/org/eclipse/lsp/cobol/core/model/variables/IndexItem.java [hunk 1]
```diff
@@ -31,14 +31,14 @@
...
     // IndexItem is non global when it's initially created.
     // Later on it will be recreated via {@code} updateParent {@code}.
-    super(levelNumber, name, definition, false, null);
+    super(-1, name, definition, false, null);
   }
 
-  public IndexItem(int leverNumber, String name, Locality definition, Variable parent) {
-    super(leverNumber, name, definition, parent);
+  public IndexItem(String name, Locality definition, Variable parent) {
+    super(-1, name, definition, parent);
```
Comment (asatklichov) [inline server/src/main/java/org/eclipse/lsp/cobol/core/model/variables/IndexItem.java:41]: I was expecting meaningful const for this LEVEL.

**Review by asatklichov — APPROVED** [review]
🐻

**Review by asatklichov — APPROVED** [review]
🦓

**Review by grianbrcom — APPROVED** [review]
:rocket:

**Pull request merged**